### PR TITLE
Fix operator crash for non-string outputs

### LINKED ISSUE #12: Non-string output crashes operator
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Adding a non-string output to a Workspace resource will cause the operator to crash.

### terraform-k8s & Kubernetes Version

terraform-k8s: 0.1.0-alpha
kubernetes: v1.17.0

### Affected Resource(s)

`Workspace`

### terraform-k8s Configuration Files

```yaml
apiVersion: app.terraform.io/v1alpha1
kind: Workspace
metadata:
  name: dns-zone-whatever-com
spec:
  organization: our-org
  secretsMountPath: "/tmp/secrets"
  module:
    source: "git::https://github.com/ryanewk/tfe-controller-terraform-modules.git//google-dns-zone"
  outputs:
  - key: zone_id
    moduleOutputName: id
  - key: name_servers
    moduleOutputName: name_servers
  variables:
  - key: project
    valueFrom:
      configMapKeyRef:
        name: dns-zone-whatever-com
        key: project
    sensitive: false
    environmentVariable: false
  - key: name
    value: whatever
    sensitive: false
    environmentVariable: false
  - key: dns_name
    value: whatever.com.
    sensitive: false
    environmentVariable: false
  - key: GOOGLE_CREDENTIALS
    sensitive: true
    environmentVariable: true
  - key: CONFIRM_DESTROY
    value: "1"
    sensitive: false
    environmentVariable: true
```

### Debug Output

https://gist.github.com/ryanewk/3421c385de0c2a3aa8441c85d9dbc4f9

### Expected Behavior

If non-string outputs are not supported, the operator should catch and log the error but it should not crash.

### Actual Behavior

Operator crashes and cannot restart:

```
$ kubectl get pod
NAME                                                   READY   STATUS             RESTARTS   AGE
tfoperator-terraform-sync-workspace-6444c44755-dgtwc   0/1     CrashLoopBackOff   36         3h8m
```

### Steps to Reproduce

See above Workspace resource. The presence of the `name_servers` output causes the operator to crash.

### Important Factoids

This is a bug report but also a feature request. It would be very useful to us if the operator supported non-string outputs.

### PR DESCRIPTION
**Pull request by joatmon08**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12 

Release note for [CHANGELOG](https://github.com/hashicorp/terraform-k8s/blob/master/CHANGELOG.md):

```release-note
- Handle non-string Terraform outputs and return them as JSON-formatted string
```

Output:

```
$ kubectl describe workspace -n preprod salutations
Status:
  Outputs:
    Key:         queue_urls
    Value:       [{"dlq":false,"url":"https://sqs.us-east-1.amazonaws.com/REDACTED/goodbye"},{"dlq":true,"url":"https://sqs.us-east-1.amazonaws.com/REDACTED/goodbye-dlq"}]
  Run ID:        run-REDACTED
  Run Status:    applied
  Workspace ID:  ws-REDACTED
```

```
$ kubectl describe configmap -n preprod salutations-outputs                                                                                                                                            

Data
====
queue_urls:
----
[{"dlq":false,"url":"https://sqs.us-east-1.amazonaws.com/REDACTED/goodbye"},{"dlq":true,"url":"https://sqs.us-east-1.amazonaws.com/REDACTED/goodbye-dlq"}]
```

### CODE REVIEW

File: operator/pkg/controller/workspace/tfc_output.go [hunk 1]
```diff
@@ -18,6 +21,92 @@ func (t *TerraformCloudClient) GetStateVersionDownloadURL(workspaceID string) (s
...
+	case ty.IsPrimitiveType():
+		switch ty {
+		case cty.String:
+			{
+				// Special behavior for JSON strings containing array or object
+				src := []byte(val.AsString())
+				ty, err := ctyjson.ImpliedType(src)
+				// check for the special case of "null", which decodes to nil,
+				// and just allow it to be printed out directly
+				if err == nil && !ty.IsPrimitiveType() && strings.TrimSpace(val.AsString()) != "null" {
```
Comment (aareet) [inline operator/pkg/controller/workspace/tfc_output.go]: Is it worth adding a test for this null branch since we're dealing with a special case?
